mav_trajectory_generation_example/src/example_planner.cc

```diff
@@ -15,129 +15,125 @@ ExamplePlanner::ExamplePlanner(ros::NodeHandle& nh) :
+  start.makeStartOrEnd(current_pose_.translation(),
                        derivative_to_optimize);
+
+  // set start point's velocity to be constrained to current velocity
+  start.addConstraint(mav_trajectory_generation::derivative_order::VELOCITY,
+                      current_velocity_);
+
+  // add waypoint to list
   vertices.push_back(start);
 
-  // set middle point constraints
-  Eigen::Vector3d middle_point_position = current_position_.translation();
-
-  // if we are above safety threshold, this is only an intermediate position
-  if (above_safety) {
-    middle_point_position.z() += safety_altitude_;
-    middle.addConstraint(mav_trajectory_generation::derivative_order::POSITION,
-                         middle_point_position);
-
-    Eigen::Vector3d middle_point_velocity;
-    middle_point_velocity << 0.0, 0.0, max_v_ * 0.1;
-    middle.addConstraint(mav_trajectory_generation::derivative_order::VELOCITY,
-                         middle_point_velocity);
-  } else {
-    // if not, this is the final waypoint.
-    middle_point_position.z() = target_altitude;
-    middle.makeStartOrEnd(middle_point_position, derivative_to_optimize);
-  }
-  vertices.push_back(middle);
-
-  // plan final point if needed (to end position at rest).
-  if (above_safety) {
-    Eigen::Vector3d end_point_position = current_position_.translation();
-    end_point_position.z() = target_altitude;
-    end.makeStartOrEnd(end_point_position, derivative_to_optimize);
-    vertices.push_back(end);
-  }
-
-
-  // compute segment times
-  std::vector<double> segment_times;
-  segment_times = estimateSegmentTimes(vertices, max_v_, max_a_);
 
+  /******* Configure end point *******/
+  // set end point constraints to desired position and set all derivatives to zero
+  end.makeStartOrEnd(goal_pos,
+                     derivative_to_optimize);
 
-  // compute segment times at mach slower speed / acceleration and use for
-  // first segment.
-  // a bit hacky....
-  std::vector<double> segment_times_slow;
-  segment_times_slow =
-      estimateSegmentTimes(vertices, max_v_ * 0.2, max_a_ * 0.2);
-  segment_times[0] = segment_times_slow[0];
+  // set start point's velocity to be constrained to current velocity
+  end.addConstraint(mav_trajectory_generation::derivative_order::VELOCITY,
+                    goal_vel);
 
+  // add waypoint to list
+  vertices.push_back(end);
 
-  // solve trajectory
+  // setimate initial segment times
+  std::vector<double> segment_times;
+  const double v_max = 2.0;
+  const double a_max = 2.0;
+  segment_times = estimateSegmentTimes(vertices, v_max, a_max);
+
+  // Set up polynomial solver with default params
+  mav_trajectory_generation::NonlinearOptimizationParameters parameters;
+  parameters.max_iterations = 1000;
+  parameters.f_rel = 0.05;
+  parameters.x_rel = 0.1;
+  parameters.time_penalty = 500.0;
+  parameters.initial_stepsize_rel = 0.1;
+  parameters.inequality_constraint_tolerance = 0.1;
```

Comment:
I think these are more or less the default values. Don't need to set them separately here imo

Reply:
:+1: